pyresample/geometry.py

```diff
@@ -1423,10 +1901,119 @@ def copy(self, **override_kwargs):
 
     def aggregate(self, **dims):
         """Return an aggregated version of the area."""
+        warnings.warn("'aggregate' is deprecated, use 'downsample' or 'upsample' instead.", PendingDeprecationWarning)
         width = int(self.width / dims.get('x', 1))
         height = int(self.height / dims.get('y', 1))
         return self.copy(height=height, width=width)
 
+    def downsample(self, x=1, y=1):
+        """Return a downsampled version of the area."""
+        # Check input validity
+        x = int(x)
+        y = int(y)
+        if x == 1 and y == 1:
+            return self
+        if x < 1 or y < 1:
+            raise ValueError('x and y arguments must be positive integers larger or equal to 1.')
+        # Downsample
+        width = int(self.width / x)
+        height = int(self.height / y)
+        return self.copy(height=height, width=width)
+
+    def upsample(self, x=1, y=1):
+        """Return an upsampled version of the area."""
+        # Check input validity
+        x = int(x)
+        y = int(y)
+        if x == 1 and y == 1:
+            return self
+        if x < 1 or y < 1:
+            raise ValueError('x and y arguments must be positive integers larger or equal to 1.')
+        # Upsample
+        width = int(self.width * x)
+        height = int(self.height * y)
+        return self.copy(height=height, width=width)
+
+    def extend(self, x=0, y=0):
+        """Extend AreaDef by x/y pixels on each side."""
+        if self.is_geostationary:
+            raise NotImplementedError("'extend' method is not implemented for GEO AreaDefinition.")
```

Comment:
why can't the area extents of a geos area def be extended?

Reply:
Lons/lats arrays of geostationary area are a 2D matrix with a disk surrounded by nan or inf. 
By modifying width and height, we don't get the desired extended/shrunk area, but we just add/drop row/ column of NaN/Inf.

Reply:
I also replaced `x` and `y` arguments with `left, right, bottom, top` arguments to be consistent with `SwathDefinition` methods

Reply:
Also, `left, right, bottom, top` make not lot of sense with geos disc area ...